memory/unix/apr_pools.c

```diff
@@ -828,12 +840,20 @@ APR_DECLARE(void *) apr_palloc(apr_pool_t *pool, apr_size_t in_size)
     apr_memnode_t *active, *node;
     void *mem;
     apr_size_t size, free_index;
+#ifdef __CHERI_PURE_CAPABILITY__
+    apr_size_t cheri_align;
+    char *first_avail;
+#endif
 
     pool_concurrency_set_used(pool);
     size = APR_ALIGN_DEFAULT(in_size);
 #if HAVE_VALGRIND
     if (apr_running_on_valgrind)
         size += 2 * REDZONE;
+#endif
+#ifdef __CHERI_PURE_CAPABILITY__
+    size = __builtin_cheri_round_representable_length(size);
+    cheri_align = ~__builtin_cheri_representable_alignment_mask(size) + 1;
```

Comment:
Hmm I wonder if cheriintrin.h should also provide
```
#define cheri_representable_alignment(size) (~cheri_representable_alignment_mask(size) + 1)
```

Reply:
It might indeed be useful as this is a common pattern.